coderd/gitauth/oauth.go

```diff
@@ -45,13 +45,13 @@ var scope = map[codersdk.GitProvider][]string{
 	codersdk.GitProviderGitHub: {"repo", "workflow"},
 }
 
-// regex provides defaults for each Git provider to
-// match their SaaS host URL. This is configurable by each provider.
+// regex provides defaults for each Git provider to match their SaaS host URL.
+// This is configurable by each provider.
 var regex = map[codersdk.GitProvider]*regexp.Regexp{
-	codersdk.GitProviderAzureDevops: regexp.MustCompile(`dev\.azure\.com`),
-	codersdk.GitProviderBitBucket:   regexp.MustCompile(`bitbucket\.org`),
-	codersdk.GitProviderGitLab:      regexp.MustCompile(`gitlab\.com`),
-	codersdk.GitProviderGitHub:      regexp.MustCompile(`github\.com`),
+	codersdk.GitProviderAzureDevops: regexp.MustCompile(`^(https?:\/\/)?dev\.azure\.com(\/.*)?$`),
```

Comment:
Does forward slash need to be escaped here? It’s often a separator in regexes, requiring the escape, but unsure if needed here.

Reply:
I think you're right, I'll remove the slashes